slice-test/slice-mapper-test/src/test/groovy/com/cognifide/slice/mapper/annotation/PreMappingTest.groovy

```diff
@@ -0,0 +1,44 @@
+/*-
+ * #%L
+ * Slice - Mapper Tests
+ * %%
+ * Copyright (C) 2012 Cognifide Limited
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.cognifide.slice.mapper.annotation
+
+import com.cognifide.slice.test.setup.BaseSetup
+import org.junit.Assert
+
+/**
+ * @author Krzysztof Watral
+ */
+class PreMappingTest extends BaseSetup {
+
+	def "PreMapping Annotation Test"() {
+		setup: "Creating initial content"
+		pageBuilder.content {
+			test("cq:PageContent") { "jcr:content"("field": "some value") }
+		}
+
+		when: "Get a model instance by path"
+		PreMappingModel preMappingModel = modelProvider.get(PreMappingModel.class, "/content/test/jcr:content")
+
+		then: "Model has been property initialized"
```

Comment:
Some weird grammar here. Did you mean `Model has been properly initialised` or `The model's property has been initialised`?

Also, maybe I'm being picky but it would be nice to describe what _proper initalisation_ means in this case (i.e. that the annotated method has been executed)

Do we want to verify that the method was called exactly once or in a specific order or is that an overkill?